ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/AvroDataTranslatorTest.java

```diff
@@ -288,4 +292,37 @@ public void shouldDropOptionalFromRootStructSchema() {
     // Then:
     assertThat("Root required", translator.getAvroCompatibleSchema().isOptional(), is(false));
   }
+
+  @Test
+  public void shouldRejectUnmatchingDecimalSchema() {
+    // Given:
+    final Schema ksqlSchema = DecimalUtil.builder(4, 2).build();
+    final Schema topicSchema = DecimalUtil.builder(6, 3).build();
+
+    // When:
+    final AvroDataTranslator translator =
+        new AvroDataTranslator(ksqlSchema, AvroProperties.DEFAULT_AVRO_SCHEMA_FULL_NAME);
+
+    // Then:
+    final ArithmeticException e = assertThrows(
+        ArithmeticException.class,
+        () -> translator.toKsqlRow(topicSchema, new BigDecimal("123.456")));
+    assertThat(e.getMessage(), containsString("A field with precision 4 and scale 2 must round"));
```

Comment:
Could we include more text here for verification to show the max value comparison, such like:
```
Numeric field overflow: A field with precision 4 and scale 2 must round to an absolute value less than 10^4
```